### PR TITLE
[mono] Fix the decoding of runtime invoke wrappers of type WRAPPER_SU…

### LINKED ISSUE #96041: [mono][tests] System.Numerics.Vectors.Tests attempting to JIT compile method
### Error Blob

The `System.Numerics.Vectors.Tests` tests are failing on Apple mobile platforms with the following error message: 
```
Attempting to JIT compile method '(wrapper runtime-invoke) object <Module>:runtime_invoke_void__this___single_single_single_single_single_single_single_single_single_single_single_single_single_single_single_single (object,intptr,intptr,intptr)' while running in aot-only mode
```

The runtime can't find the wrapper method for the test case:
https://github.com/dotnet/runtime/blob/9a15fa9371ca786c5db92336090c994d7f4bfe14/src/libraries/System.Numerics.Vectors/tests/Matrix4x4Tests.cs#L44-L72

The issue has been introduced in https://github.com/dotnet/runtime/pull/95523. It looks like the runtime doesn't get the correct runtime invoke wrapper for `WRAPPER_SUBTYPE_RUNTIME_INVOKE_NORMAL`.

Build example: https://dev.azure.com/dnceng-public/public/_build/results?buildId=499215&view=logs&j=acac6bf3-3edd-5022-4e45-3362a02520b1&t=1de8bf71-8fe5-5d5c-4360-8c2eb159d012

```json
{
  "ErrorMessage": "Attempting to JIT compile method",
  "BuildRetry": false,
  "ErrorPattern": "",
  "ExcludeConsoleLog": true
}
```

### Reproduction Steps

1. Build the runtime for iossimulator: `/build.sh mono+libs -os iossimulator -arch arm64 -c Debug`
2. Build the tests locally: `./dotnet.sh build /t:Test -c Debug src/libraries/System.Numerics.Vectors/tests/System.Numerics.Vectors.Tests.csproj /p:TargetOS=iossimulator /p:TargetArchitecture=arm64 /p:RunAOTCompilation=true /p:MonoForceInterpreter=false /p:MonoEnableLLVM=true`
3. Run the tests using Xcode

 ### Known issue validation
**Build: :mag_right:** 
**Result validation: :warning:** Validation could not be done without an Azure DevOps build URL on the issue. Please add it to the "**Build: :mag_right:**" line.
**Validation performed at:** 12/14/2023 9:25:18 PM UTC

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[499343](https://dev.azure.com/dnceng-public/public/_build/results?buildId=499343)|dotnet/runtime|[baseservices/varargs/varargsupport_r/varargsupport_r.cmd](https://dev.azure.com/dnceng-public/public/_build/results?buildId=499343&view=ms.vss-test-web.build-test-results-tab&runId=11618000&resultId=100029)|dotnet/runtime#92057|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|1|1|

### PR DESCRIPTION
**Pull request by vargaz**
…BTYPE_RUNTIME_INVOKE_NORMAL.

Fixes https://github.com/dotnet/runtime/issues/96041.

### CODE REVIEW

File: src/mono/mono/mini/aot-runtime.c [hunk 1]
```diff
@@ -1216,7 +1216,27 @@ decode_method_ref_with_target (MonoAotModule *module, MethodRef *ref, MonoMethod
 			}
 			case WRAPPER_SUBTYPE_RUNTIME_INVOKE_NORMAL: {
 				MonoMethodSignature *sig = decode_signature_with_target (module, NULL, p, &p);
```
Comment (kotlarmilos) [inline src/mono/mono/mini/aot-runtime.c:1218]:
```suggestion
```

File: src/mono/mono/mini/aot-runtime.c [hunk 2]
```diff
@@ -1216,7 +1216,27 @@ decode_method_ref_with_target (MonoAotModule *module, MethodRef *ref, MonoMethod
...
-				ref->method = mono_marshal_get_runtime_invoke_for_sig (sig);
+
+				/*
+				 * Can't decode sig into a runtime invoke wrapper since mono_marshal_get_runtime_invoke_for_sig () returns
+				 * a different method right now.
+				 */
+				if (!target)
+					return FALSE;
+				info = mono_marshal_get_wrapper_info (target);
```
Comment (kotlarmilos) [inline src/mono/mono/mini/aot-runtime.c:1226]:
```suggestion
				MonoMethodSignature *sig = decode_signature_with_target (module, NULL, p, &p);
				info = mono_marshal_get_wrapper_info (target);
```

**Review by kotlarmilos — COMMENTED** [review]
LGTM! I suggest to first check `if (!target)` to avoid memory leaks with `sig`.

**Review by kotlarmilos — APPROVED** [review]

**Comment by vargaz** [comment]
Failures are relevant.

**Comment by vargaz** [comment]
This will not work, it assumes that when decoding runtime invoke wrappers,target!=NULL, which is not always true.

**Pull request closed without merging**